dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -164,6 +164,7 @@ public Version parseWorkflowContent(String filePath, String content, Set<SourceF
                     version.setDescriptionAndDescriptionSource(finalChoiceForDescription, DescriptionSource.DESCRIPTOR);
                 } else {
                     LOG.info("Description not found!");
+                    version.setDescriptionAndDescriptionSource(null, null);
```

Comment:
Rather than have to modify the language handlers when something like this comes up, could we "reset" the existing `Version`/`VersionMetadata` object early in the parse, and then add the information back again when we invoke the language handlers?

Going even further, I could even imagine a function like `setDescriptionAndDescriptionSource` comparing the new source to the existing source, and only overwriting the description if the source was "better".  That way, the code could blithely call  `setDescriptionAndDescriptionSource`, and `setDescriptionAndDescriptionSource` would decide the precedence.  You'd need a separate `resetDescriptionAndDescriptionSource` function, though.
 and it would decide

Reply:
I ended up removing the code that resets the description and description source in the language handlers.

After merging develop with this branch, there was a change where this resetting was no longer needed because the version metadata is now being set on the new remote version (which has the empty values) then fields of the remote version are copied over to the existing version ([code](https://github.com/dockstore/dockstore/blob/30c353347ee09b06e6aad9e594ba4a07e731675c/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java#L760-L761))